src/strategies/MangroveOrder.sol

```diff
@@ -46,13 +46,18 @@ contract MangroveOrder is Forwarder, IOrderLogic {
     }
   }
 
+  //  added this event to keep track of what the expiry date is.
+  event SetExpiry(address indexed outbound_tkn, address indexed inbound_tkn, uint offerId, uint date);
```

Comment:
hm is it because we need a history? I am a bit torn about emitting events for every single change - we could just read this value from the chain, I would not expect us to use the indexer for every read.

Reply:
This is the same principle that is followed in Mangrove: All interesting state changes are signaled via events.

Reply:
If you run a cleaning bot you want to maintain expiry dates in your cache.